src/ckb-daemon/input.c

```diff
@@ -12,39 +13,154 @@ int macromask(const uchar* key1, const uchar* key2){
     return 1;
 }
 
+///
+/// \brief pt_head is the head pointer for the single linked thread list managed by macro_pt_en/dequeue().
+static ptlist_t* pt_head = 0;
+/// \brief pt_tail is the tail pointer for the single linked thread list managed by macro_pt_en/dequeue().
+static ptlist_t* pt_tail = 0;
+
+///
+/// \brief macro_pt_enqueue Save the new thread in the single linked list (FIFO).
```

Comment:
I had looked for a solution like this, but not long enough (I just found other self implementations). So thank you for the link.
Because Tatokis asked me to merge the implementation before 20th june, I will change it as a PR after this date.